rosbag2_transport/test/rosbag2_transport/test_play.cpp

```diff
@@ -554,6 +554,164 @@ TEST_P(RosBag2PlayTestFixtureMessageOrder, recorded_msgs_are_played_for_all_topi
+    (msg->int32_value += 2, serialize_test_message("topic1", 5, 4, msg)),
+    (msg->int32_value += 2, serialize_test_message("topic3", 7, 7, msg)),
+    (msg->int32_value += 2, serialize_test_message("topic1", 9, 8, msg)),
+    (msg->int32_value += 2, serialize_test_message("topic3", 11, 14, msg))
+  });
+  msg->int32_value = 2;
+  messages_list.emplace_back(std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>>{
+    serialize_test_message("topic2", 1, 1, msg),
+    (msg->int32_value += 2, serialize_test_message("topic4", 3, 2, msg)),
+    // (msg->int32_value += 2, serialize_test_message("topic1", 5, 4, msg)),
+    (msg->int32_value += 2, serialize_test_message("topic2", 5, 4, msg)),
+    (msg->int32_value += 2, serialize_test_message("topic4", 7, 7, msg)),
+    (msg->int32_value += 2, serialize_test_message("topic2", 9, 9, msg)),
+    (msg->int32_value += 2, serialize_test_message("topic4", 11, 14, msg))
+  });
+  std::vector<rosbag2_transport::Player::reader_storage_options_pair_t> bags{};
+  std::size_t total_messages = 0u;
+  for (const auto & curr_bag_messages : messages_list) {
+    auto prepared_mock_reader = std::make_unique<MockSequentialReader>();
+    total_messages += curr_bag_messages.size();
+    prepared_mock_reader->prepare(curr_bag_messages, topic_types);
+    bags.emplace_back(
+      std::make_unique<rosbag2_cpp::Reader>(std::move(prepared_mock_reader)), storage_options_);
+  }
+  ASSERT_GT(total_messages, 0u);
+  const rosbag2_transport::MessageOrder message_order = GetParam();
+  play_options_.message_order = message_order;
+  play_options_.read_ahead_queue_size = total_messages - 3;
+  auto player = std::make_shared<rosbag2_transport::Player>(std::move(bags), play_options_);
+  std::size_t num_played_messages = 0u;
+  rcutils_time_point_value_t last_timestamp = 0;
+  const auto get_timestamp =
+    [message_order](std::shared_ptr<rosbag2_storage::SerializedBagMessage> msg) {
+      switch (message_order) {
+        case rosbag2_transport::MessageOrder::RECEIVED_TIMESTAMP:
+          return msg->recv_timestamp;
+        case rosbag2_transport::MessageOrder::SENT_TIMESTAMP:
+          return msg->send_timestamp;
+        default:
+          throw std::runtime_error("unknown rosbag2_transport::MessageOrder value");
+      }
+    };
+  const auto callback = [&](std::shared_ptr<rosbag2_storage::SerializedBagMessage> playing_msg) {
+      // Make sure messages are played in order
+      num_played_messages++;
+      const auto timestamp = get_timestamp(playing_msg);
+      using MessageT = typename decltype(msg)::element_type;
+      const auto deserialized_msg = deserialize_test_message<MessageT>(playing_msg);
+      // The int32_value was set in an increasing order when creating the messages
+      // std::cout << "deserialized_value = " << deserialized_msg->int32_value << ", timestamp = "
+      //   << RCUTILS_NS_TO_MS(timestamp) << ", topic_name = `" << playing_msg->topic_name << "`\n";
```

Comment:
Should these lines be deleted?

Reply:
I would like to keep it. It will be very useful to quickly debug the failing test.